src/three-types.ts

```diff
@@ -12,10 +12,10 @@ export type Overwrite<T, O> = Omit<T, NonFunctionKeys<O>> & O
 type Args<T> = T extends new (...args: any) => any ? ConstructorParameters<T> : T
 
 export declare namespace ReactThreeFiber {
-  type Euler = THREE.Euler | number[]
-  type Matrix4 = THREE.Matrix4 | number[]
-  type Vector3 = THREE.Vector3 | number[]
-  type Color = THREE.Color | number | string
+  type Euler = THREE.Euler | Parameters<THREE.Euler['set']>
```

Comment:
> Parameters<THREE.Euler['set']>

Nice one! I didn't get it when I read your message for the first time, but now it fits perfectly.

![image](https://user-images.githubusercontent.com/15332326/75493744-e93b7b00-59ba-11ea-89c7-4ecd151fe18b.png)

This is way better than `number[]`!